src/osemosys.txt

```diff
@@ -56,11 +56,18 @@ set STORAGE;
 param ResultsPath, symbolic default 'results';
 param YearSplit{l in TIMESLICE, y in YEAR};
 
-param DiscountRate{r in REGION, t in TECHNOLOGY};
-param CapitalRecoveryFactor{r in REGION, t in TECHNOLOGY, y in YEAR} :=
-	(1 + DiscountRate[r, t]) ^ (y - min{yy in YEAR} min(yy) + 0.0);
-param CapitalRecoveryFactorMid{r in REGION, t in TECHNOLOGY, y in YEAR} :=
-	(1 + DiscountRate[r, t]) ^ (y - min{yy in YEAR} min(yy) + 0.5);
+param DiscountRate{r in REGION};
+param DiscountRateIdv{r in REGION, t in TECHNOLOGY};
+
+param CapitalRecoveryFactor{r in REGION, y in YEAR} :=
+	(1 + DiscountRate[r]) ^ (y - min{yy in YEAR} min(yy) + 0.0);
+param CapitalRecoveryFactorMid{r in REGION, y in YEAR} :=
+	(1 + DiscountRate[r]) ^ (y - min{yy in YEAR} min(yy) + 0.5);
+
+param OperationalLife{r in REGION, t in TECHNOLOGY};
+
+param CapitalRecoveryFactorIdv{r in REGION, t in TECHNOLOGY, y in YEAR} :=  
+	(DiscountRateIdv[r,t] / (1 - (1 + DiscountRateIdv[r,t])^(-(OperationalLife[r,t])))) * ((1 - (1 + DiscountRateIdv[r,t])^(-(OperationalLife[r,t]))) * (1 + DiscountRateIdv[r,t]) / DiscountRateIdv[r,t]) / ((1 + DiscountRate[r]) ^ (y - min{yy in YEAR} min(yy) + 0.0));
```

Comment:
Couldn't this equation be simplified as below?
`(1 + DiscountRateIdv[r,t]) / ((1 + DiscountRate[r]) ^ (y - min{yy in YEAR} min(yy) + 0.0))`

Reply:
Hi, Abhi, thanks for testing the script. Yes, it is possible to simplify that eq.. with (1+DR). I left it as a long eq. , so others can see where the different factors come from.